server/graphql/v2/mutation/MemberMutations.js

```diff
@@ -0,0 +1,56 @@
+import { GraphQLNonNull, GraphQLString } from 'graphql';
+
+import models from '../../../models';
+import { NotFound, Unauthorized } from '../../errors';
+import { idDecode } from '../identifiers';
+import { AccountReferenceInput } from '../input/AccountReferenceInput';
+import { Member } from '../object/Member';
+
+const memberMutations = {
+  editPublicMessage: {
+    type: new GraphQLNonNull(Member),
+    description: 'Edit the public message for the given Member of a Collective',
+    args: {
+      FromCollectiveLegacyId: {
+        type: AccountReferenceInput,
+        description: 'Legacy id for the donating Collective',
+      },
+      ToCollectiveId: {
+        type: AccountReferenceInput,
+        description: 'Public id for the receiving Collective',
+      },
+      message: {
+        type: GraphQLString,
+        description: 'New public message',
+      },
+    },
+    async resolve(_, args, req) {
+      const { FromCollectiveLegacyId, ToCollectiveId, message } = args;
+
+      if (!req.remoteUser || !req.remoteUser.isAdmin(FromCollectiveLegacyId)) {
```

Comment:
`FromCollectiveLegacyId` is an account reference, doing `req.remoteUser.isAdmin(FromCollectiveLegacyId)` won't work